src/armor_layers.cpp

```diff
@@ -746,114 +807,85 @@ void player::sort_armor()
-                        inv.reassign_item( w, invlet );
-                        ++witer;
-                        ++iiter;
-                    }
+        } else if( action == "DROP" ) { //same as TAKE_OFF
+            if( g->u.query_yn( _( "Drop selected armor?" ) ) ) {
+                do_return_entry();
+                auto cur_item_iter = cur_grp->get_cur_item_info().get_iter();
+                drop( get_item_position( &*cur_item_iter ), pos() );
+                if( !g->u.has_activity( activity_id( "ACT_ARMOR_LAYERS" ) ) ) {
+                    return;
                 }
+                g->u.cancel_activity();
             }
-        } else if( action == "USAGE_HELP" ) {
-            popup_getkey( _( "\
-Use the arrow- or keypad keys to navigate the left list.\n\
-[%s] to select highlighted armor for reordering.\n\
-[%s] / [%s] to scroll the right list.\n\
-[%s] to assign special inventory letters to clothing.\n\
-[%s] to change the side on which item is worn.\n\
-[%s] to sort armor into natural layer order.\n\
-[%s] to equip a new item.\n\
-[%s] to equip a new item at the currently selected position.\n\
-[%s] to remove selected armor from oneself.\n\
- \n\
-[Encumbrance and Warmth] explanation:\n\
-The first number is the summed encumbrance from all clothing on that bodypart.\n\
-The second number is an additional encumbrance penalty caused by wearing multiple items \
-on one of the bodypart's layers or wearing items outside of other items they would \
-normally be work beneath (e.g. a shirt over a backpack).\n\
-The sum of these values is the effective encumbrance value your character has for that bodypart." ),
-                          ctxt.get_desc( "MOVE_ARMOR" ),
-                          ctxt.get_desc( "PREV_TAB" ),
-                          ctxt.get_desc( "NEXT_TAB" ),
-                          ctxt.get_desc( "ASSIGN_INVLETS" ),
-                          ctxt.get_desc( "CHANGE_SIDE" ),
-                          ctxt.get_desc( "SORT_ARMOR" ),
-                          ctxt.get_desc( "EQUIP_ARMOR" ),
-                          ctxt.get_desc( "EQUIP_ARMOR_HERE" ),
-                          ctxt.get_desc( "REMOVE_ARMOR" )
-                        );
-            draw_grid( w_sort_armor, left_w, middle_w );
-        } else if( action == "HELP_KEYBINDINGS" ) {
-            draw_grid( w_sort_armor, left_w, middle_w );
+        }
+
+        else if( action == "USAGE_HELP" ) {
+            std::string help_str;
+            std::string head_color = string_from_color( c_white );
+            std::string desc_color = string_from_color( c_light_gray );
+
+            for( size_t i = 0; i < column_num_entries; i++ ) {
+                sort_armor_col_data col_data = get_col_data( i );
+                if( static_cast<sort_armor_columns>( i ) == column_divider ) {
+                    help_str += "\n";
+                } else {
+                    help_str += string_format( "<color_%s>%s</color> - <color_%s>%s</color>\n",
```

Comment:
```suggestion
                    help_str += colorize( col_data.get_name(), c_white );
                    help_str += " - ";
                    help_str += colorize( col_data.get_description(), c_light_gray );
                    help_str += "\n";
```
Generally using the `colorize` function is better than adding the color tags manually.